### DISCUSSION TITLE
Lag spikes upon playing for the first time

### DISCUSSION THREAD

**Discussion by Blahajmoder** [ppy/osu#22514] [Q&A]
### Type

Performance

### Bug description

Whenever I play the game for the first time i have massive lag spikes on the first play of any map after starting the game. After I restart it gets better though, but still i get small lag spikes. It also happens when I change skins too and it's very noticeable at the first few seconds. They aren't making the game unplayable but they aren't small enough either, it is at a point where it's very annoying to play because often i have misses that are because of this. I noticed this happening after I tried to merge my stable and Lazer folder. 

### Version

2023.123.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/10579852/database.log)
[input.log](https://github.com/ppy/osu/files/10579853/input.log)
[network.log](https://github.com/ppy/osu/files/10579854/network.log)
[performance.log](https://github.com/ppy/osu/files/10579855/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10579856/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10579857/updater.log)

**Comment by Shirou27** [comment]
We're same. I experienced this very often. Every first play of a map has a lag spike but when I retry, the lag spike stops.

I play in a phone btw.

**Reply by bdach** [reply]
phone performance is not relevant in this thread and is a constant work in progress.

**Comment by bdach** [comment]
Please post video of this, with frame graph showing (ctrl-f11 twice).

**Reply by Blahajmoder** [reply]
https://user-images.githubusercontent.com/124374214/216780968-54859126-21fa-4857-9ff2-573fb12955d9.mp4

Example at the start of the Beatmap and at 0:20.
Note: I did a recording with and without discord overlay and it didn't change anything. The call overlay is just a visual bug and I am not actually streaming.

**Reply by bdach** [reply]
Looks potentially related to the skin; do you mind uploading it? Logs from this run would also be nice if you still have them.

Also, HDD or SSD drive?

**Reply by Blahajmoder** [reply]
**Logs:**
[input.log](https://github.com/ppy/osu/files/10609265/input.log)
[network.log](https://github.com/ppy/osu/files/10609267/network.log)
[performance.log](https://github.com/ppy/osu/files/10609268/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10609269/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10609270/updater.log)
[database.log](https://github.com/ppy/osu/files/10609271/database.log)
**The skin:**
[#Azerite (osu player84) #2.zip](https://github.com/ppy/osu/files/10609284/Azerite.osu.player84.2.zip)


I don't think it is because of the skin because it happens with all of my skins. I just couldn't include more because of the 10MB limit.
I play on an HDD but it doesn't happen on stable at all, only on Lazer. It also didn't happen until the update with the osu! folder merging.
